Replace html2markdown with to-markdown

epsil opened the pull request:
I have ported clipboard2markdown from [html2markdown](https://github.com/kates/html2markdown) to [to-markdown](https://github.com/domchristie/to-markdown). This converter is *much* easier to customize and extend.

I have added some code to tailor the converted Markdown to the guidelines described at <http://pandoc.org/README.html#pandocs-markdown>.

I have also simplified the HTML interface a little, and updated the instructions.

euangoddard commented:
I'm very happy to merge this pull request, but it seems like your work is quite extensive. Would you not rather consider this as a separate thing given all your hard work?

Let me know and if you're happy for it to be an iteration on the original I'll merge it!

epsil commented:
I started out by hacking on the `html2markdown.js` file, adding additional Markdown features as needed. However, I quickly found it cumbersome to keep track of state between html2markdown's `start` and `end` functions. Then I found [to-markdown](https://github.com/domchristie/to-markdown), which is written in a much more composable style. I replaced the converter and ported my own changes over, which was quick work, and added a few more changes (see the beginning of `clipboard2markdown.js`). I have followed the style guidelines used by [pandoc's Markdown](https://github.com/domchristie/to-markdown/pull/149), which is a mature implementation.

If you like it, please merge it. Credit is not that important to me; what *is* important is that the code be made available where others can find it and contribute to it. I found your code by accident, by the way, as I googled for "paste to markdown". This turned up a few `xclip`-to-`pandoc` scripts, but little more. To help out Google, I have padded `README.md` with some additional rephrasings.

colcord commented:
Hi Epsil, I just saw all the hard work you did helping this project move to to-markdown. Is there any chance you could help with the migration to turndown? I've logged an issue, but since I'm not a coder, I can't make the kind of changes you did to help. kind regards, Frank
https://github.com/euangoddard/clipboard2markdown/issues/3
